File: frontend/src/api/orderService.js
import axiosClient from './axiosClient';

/**
 * Shopping Order API Service
 * Uses the shared axiosClient which automatically attaches the JWT token.
 */

/**
 * Create order from user's current shopping cart
 * @param {Object} data - { receiverName, receiverPhone, shippingAddress, note, paymentMethod }
 */
export const createOrder = async (data) => {
  const response = await axiosClient.post('/api/orders', data);
  return response.data;
};

/**
 * Get paginated list of current user's orders
 * @param {Object} params - { page, size }
 */
export const getUserOrders = async (params = {}) => {
  const response = await axiosClient.get('/api/orders', { params });
  return response.data;
};

/**
 * Get single order detail by ID
 * @param {number|string} orderId
 */
export const getOrderById = async (orderId) => {
  const response = await axiosClient.get(`/api/orders/${orderId}`);
  return response.data;
};

/**
 * Cancel an order (allowed only if status is PENDING)
 * @param {number|string} orderId
 */
export const cancelOrder = async (orderId) => {
  const response = await axiosClient.put(`/api/orders/${orderId}/cancel`);
  return response.data;
};

File: mobile/lib/services/order_service.dart
import 'package:dio/dio.dart';
import '../core/constants/api_constants.dart';
import '../core/network/dio_client.dart';
import '../models/order_model.dart';

class OrderService {
  final Dio _dio = DioClient().dio;

  /// Creates a new order from current cart
  Future<OrderModel> createOrder({
    required String receiverName,
    required String receiverPhone,
    required String shippingAddress,
    String? note,
    String paymentMethod = 'COD',
    String? voucherCode,
  }) async {
    final response = await _dio.post(
      ApiConstants.orders,
      data: {
        'receiverName': receiverName.trim(),
        'receiverPhone': receiverPhone.trim(),
        'shippingAddress': shippingAddress.trim(),
        'note': note?.trim(),
        'paymentMethod': paymentMethod,
        'voucherCode': voucherCode?.trim(),
      },
    );

    return OrderModel.fromJson(response.data as Map<String, dynamic>);
  }

  /// Gets paginated order history for current user
  Future<OrderPageResponse> getUserOrders({int page = 0, int size = 10}) async {
    final response = await _dio.get(
      ApiConstants.orders,
      queryParameters: {
        'page': page,
        'size': size,
      },
    );

    return OrderPageResponse.fromJson(response.data as Map<String, dynamic>);
  }

  /// Gets details of an order by ID
  Future<OrderModel> getOrderById(int id) async {
    final response = await _dio.get('${ApiConstants.orders}/$id');
    return OrderModel.fromJson(response.data as Map<String, dynamic>);
  }

  /// Cancels an order (PENDING only)
  Future<OrderModel> cancelOrder(int id) async {
    final response = await _dio.put('${ApiConstants.orders}/$id/cancel');
    return OrderModel.fromJson(response.data as Map<String, dynamic>);
  }
}
